shared/common-adapters/channel-container.js

```diff
@@ -0,0 +1,13 @@
+// @flow
+import Channel from './channel'
+import {connect} from '../util/container'
+import {type ConversationIDKey} from '../constants/types/chat'
+import * as ChatGen from '../actions/chat-gen'
+
+type OwnProps = {channel: string, convID: ConversationIDKey}
+
+const mapDispatchToProps = (dispatch, {convID}: OwnProps) => ({
+  onClick: () => dispatch(ChatGen.createSelectConversation({conversationIDKey: convID, fromUser: true})),
```

Comment:
This doesn't yet correctly handle the case where the current user isn't in the channel pointed to by convID.

Reply:
It would be nice if select conversation just showed you the convo in preview mode if you aren't in it, and did the normal thing if you are.

Reply:
if you can find out the user isn't in the channel around here you could do
```
dispatch(ChatGen.createPreviewChannel({conversationIDKey}))
dispatch(ChatGen.createSelectConversation({conversationIDKey, fromUser: true}))
```